gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/crud_service/integration/IntegrationCrudServiceImplTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.crud_service.integration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import fixtures.core.model.IntegrationFixture;
+import io.gravitee.apim.core.integration.model.Integration;
+import io.gravitee.apim.infra.adapter.IntegrationAdapter;
+import io.gravitee.repository.management.api.IntegrationRepository;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class IntegrationCrudServiceImplTest {
+
+    IntegrationRepository integrationRepository;
+
+    IntegrationCrudServiceImpl service;
+
+    @BeforeEach
+    void setUp() {
+        integrationRepository = mock(IntegrationRepository.class);
+        service = new IntegrationCrudServiceImpl(integrationRepository);
+    }
+
+    @Test
+    @SneakyThrows
+    void should_create_integration() {
```

Comment:
You should take a look at https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/crud_service/api_key/ApiKeyCrudServiceImplTest.java 
I think it will answer your mixed feeling about the test ;)